### PR TITLE
Add SPI Interface in yaml file and boot.config

### PR DESCRIPTION
**Pull request by tokurz**
Hey, 

I have add the SPI interface in the yaml file, I hope this is correct so, also I modified the boot.config

I'm happy when the changes can be tested in the dev part of this branch where I can download on my ubuntu core device.

Cheers

### CODE REVIEW

File: prebuilt/boot-assets/config.txt [hunk 1]
```diff
@@ -46,6 +46,8 @@ kernel=uboot.bin
 
 dtparam=i2c_arm=on
 dtparam=i2c_vc=on
+#maybe run "sudo modprobe spi-bcm2835" to load kernel modules
```
Comment (ogra1) [inline prebuilt/boot-assets/config.txt]: i dont think that comment is needed, if the kernel does not automatically load the module when the dtparam is set i would consider this a kernel bug ... can you verify if this modprobe call is actually needed ?

Comment (tokurz) [inline prebuilt/boot-assets/config.txt]: I will check that, thanks for the fast response

Comment (tokurz) [inline prebuilt/boot-assets/config.txt]: the comment is not needed just tested , its working fine with the dtparam line alone.

Cheers

**Comment by ogra1** [comment]
note that the branch changed a bit underneath this PR, the config.txt change is already in (someone needed it to work on snaps in devmode) and the snapcreaft.yaml changed slightly. Please adjust the snapcraft.yaml bit and drop the config.txt changes so we can merge this PR as soon as the interface is available in snapd.

**Comment by tokurz** [comment]
Hey @ogra1 The merge into snapcore was today successfull, 

@zyga when I can have any kind of a prebuild snap, i like to it test before ;)

https://github.com/snapcore/snapd/pull/3499

It can be here also merged then.

greets

**Comment by ogra1** [comment]
@tokurz can you resolve the conflicts ?

**Comment by tokurz** [comment]
@ogra1 how I can do that?, pull again?

**Comment by ogra1** [comment]
in your local fork of the branch:

git remote add upstream git://github.com/snapcore/pi3-gadget.git
git fetch upstream
git pull upstream master
git push

then your local tree is up to date and you can adjust, commit and push again ...

**Comment by tokurz** [comment]
Hy @ogra1 , I have merged and pushed the new snapcraft yaml file, you can now merge it.

greets

**Comment by ogra1** [comment]
@tokurz now you just need to drop the duplicated config.txt change (dtparam=spi=on is already there) ;)

**Comment by tokurz** [comment]
@ogra1 its done :)

**Pull request merged**